### PR TITLE
feat: Remove `momento account signup`

### PR DESCRIPTION
**Pull request by rlinehan**
Part of https://github.com/momentohq/user-management/issues/90.

Remove `momento account signup` command and replace with an error message pointing to the Console.

The subcommand and arguments are kept (and made options) so that if someone uses the command with these, they get a helpful error message:

```cli
$ cargo run -- account --help
Manage accounts

Usage: momento account [OPTIONS] <COMMAND>

Commands:
  signup  **DEPRECATED** Please go to the Momento Console to sign up
  help    Print this message or the help of the given subcommand(s)

Options:
      --verbose            Log more information
  -p, --profile <PROFILE>  User profile [default: default]
  -h, --help               Print help
```
```
$ cargo run -- account signup --help
**DEPRECATED** Please go to the Momento Console to sign up

Usage: momento account signup [OPTIONS] [COMMAND]

Commands:
  gcp   **DEPRECATED** Please go to the Momento Console to sign up
  aws   **DEPRECATED** Please go to the Momento Console to sign up
  help  Print this message or the help of the given subcommand(s)

Options:
      --verbose            Log more information
  -p, --profile <PROFILE>  User profile [default: default]
  -h, --help               Print help
```
```
$ cargo run -- account signup
ERROR: This command has been deprecated and removed!
Please go to the console to sign up for Momento and generate a token:
 https://console.gomomento.com
```
```
$ cargo run -- account signup aws
ERROR: This command has been deprecated and removed!
Please go to the console to sign up for Momento and generate a token:
 https://console.gomomento.com
```
```
$ cargo run -- account signup aws --email foo
ERROR: This command has been deprecated and removed!
Please go to the console to sign up for Momento and generate a token:
 https://console.gomomento.com
```

### CODE REVIEW

**Comment by rlinehan** [comment]
This is blocked on getting the new sdk version in but wanted to open it up to get feedback that this is the right approach to take.

File: momento-cli-opts/src/lib.rs [hunk 1]
```diff
@@ -117,12 +117,14 @@ pub enum SigningKeyCommand {
     List {},
 }
 
+const SIGNUP_DEPRECATED_MSG : &str = "**DEPRECATED** Please go to the Momento Console to sign up";
```
Comment (krispraws) [inline momento-cli-opts/src/lib.rs:120]: Minor nit: Deprecated suggests it still works but we prefer it is not used. Can we call it `decommissioned` or something similar?

File: momento/src/main.rs [hunk 2]
```diff
@@ -170,14 +170,19 @@ async fn run_momento_command(args: momento_cli_opts::Momento) -> Result<(), CliE
...
         momento_cli_opts::Subcommand::Account { operation } => match operation {
-            momento_cli_opts::AccountCommand::Signup { signup_operation } => match signup_operation
-            {
-                momento_cli_opts::CloudSignupCommand::Gcp { email, region } => {
-                    commands::account::signup_user(email, "gcp".to_string(), region).await?
-                }
-                momento_cli_opts::CloudSignupCommand::Aws { email, region } => {
-                    commands::account::signup_user(email, "aws".to_string(), region).await?
-                }
+            // This command has been deprecated. It now just prints out an error message.
```
Comment (krispraws) [inline momento/src/main.rs:173]: I am confused - why do we need the subcommands? Can't we get rid of `CloudSignupCommand` entirely and just print the error message here? It doesn't look like the error message varies based on aws vs gcp.

Comment (danielamiao) [reply momento/src/main.rs:173]: @rlinehan I agree with Ramya's comment and have the same question. Can't we just delete everything subcommands and keep top level command only? This is only helpful for people who may be reading outdated blog posts that are asked to run this command so as long as they can see the deprecation message we are good. Are you worried they'd get an error "subcommand doesn't exist"?

**Comment by danielamiao** [comment]
Can we add a direct URL link to the console so people can easily cut and paste into browser? I would go a step further and just pop up the browser for them with the link but that might be harder / too much effort.

**Comment by rlinehan** [comment]
> Can we add a direct URL link to the console so people can easily cut and paste into browser? I would go a step further and just pop up the browser for them with the link but that might be harder / too much effort.

@danielamiao it's in the error currently
```
ERROR: This command has been deprecated and removed!
Please go to the console to sign up for Momento and generate a token:
 https://console.gomomento.com
```

I could also do pop the browser, but personally I don't like that behavior that much? I don't think it's that hard though if you want that.

**Comment by danielamiao** [comment]
@rlinehan ah ok thanks! I think maybe just add it to help menu too?

File: momento/src/main.rs (hunk 2 above)
Comment (rlinehan) [reply momento/src/main.rs:173]: Yeah, my concern is if a user was reading old docs or something, the command they would have seen was `momento account signup aws`. If we remove the subcommands then when they run that command they'll get 

```
$ cargo run -- account signup aws
error: unexpected argument 'aws' found

Usage: momento account signup [OPTIONS]

For more information, try '--help'.
```

Although the code is annoying, it seems more friendly for users to have it give the "go to console" message for subcommands that we previously had documented.

It's possible I can clean up the code a little though, I'll look into it.

File: momento-cli-opts/src/lib.rs (hunk 1 above)
Comment (rlinehan) [reply momento-cli-opts/src/lib.rs:120]: I like decommissioned! I was thinking of removed too, but I think I like decommissioned more.

File: momento/src/main.rs (hunk 2 above)
Comment (danielamiao) [reply momento/src/main.rs:173]: I think that makes sense, there is a tradeoff between those users' experiences (small group) vs the code experience for the future devs of `momento cli` (also potentially small group). So I don't have a strong opinion either way.

Comment (krispraws) [reply momento/src/main.rs:173]: Makes sense. Thanks for the context!

Comment (rlinehan) [reply momento/src/main.rs:176]: I figured out how I could clean up the code a good amount without removing this behavior

File: momento/src/commands/account.rs [hunk 3]
```diff
@@ -1,66 +1,5 @@
...
-                    });
-                }
-            }
-            Err(e) => return Err(CliError { msg: e.to_string() }),
-        },
-        Err(e) => return Err(CliError { msg: e.to_string() }),
-    };
-    Ok(())
+pub async fn signup_deprecated() -> Result<(), CliError> {
+    return Err(CliError { msg:"This command has been deprecated and removed! \nPlease go to the console to sign up for Momento and generate a token: \n https://console.gomomento.com".to_string() });
```
Comment (krispraws) [inline momento/src/commands/account.rs:4]: s/deprecated/decommissioned

**Comment by cprice404** [comment]
can you please show what the main `help` screen looks like with these changes? I am curious whether you hid the old command and also would like to see what the current list of top level commands is. (Sorry I know I could glean this from the code but I haven't had time to go through it all yet.)

**Review by krispraws — APPROVED** [review]

**Comment by rlinehan** [comment]
> can you please show what the main help screen looks like with these changes? I am curious whether you hid the old command and also would like to see what the current list of top level commands is. (Sorry I know I could glean this from the code but I haven't had time to go through it all yet.)

@cprice404 I haven't hidden `account` from the top level of commands but I could, since `signup` is currently the only subcommand for that.

```
$ cargo run -- --help
Command line tool for Momento Serverless Cache

Usage: momento [OPTIONS] <COMMAND>

Commands:
  cache      Interact with caches
  topic      Interact with topics
  configure  Configure credentials
  account    Manage accounts
  help       Print this message or the help of the given subcommand(s)

Options:
      --verbose            Log more information
  -p, --profile <PROFILE>  User profile [default: default]
  -h, --help               Print help
  -V, --version            Print version
```

**Comment by rlinehan** [comment]
@poppoerika I pushed a commit that updates the README... do you know what the process is for updating the japanese (and also portuguese I guess) version of this?

**Review by krispraws — APPROVED** [review]

**Review by cprice404 — DISMISSED** [review]

**Review by krispraws — APPROVED** [review]

**Pull request merged**